src/modules/home/api/create-accommodation.ts

```diff
@@ -0,0 +1,23 @@
+import { useMutation } from '@tanstack/react-query';
+import type { AccommodationFormData } from '#/modules/home/types/accommodation-form-data.interface.ts';
+import { axiosInstance } from '#/configs';
+import { ENDPOINTS } from '#/modules/home/constants/endpoints.constant.ts';
+import { toast } from 'react-toastify';
+import { type AccommodationBasics } from '#/modules/home/types/accommodation-basics.type.ts';
+import { type Accommodation } from '#/modules/home/types/accommodation.type.ts';
+
+export const useCreateAccommodation = () => {
+  return useMutation({
+    mutationFn: (data: AccommodationFormData & AccommodationBasics) =>
+      axiosInstance.post<Accommodation>(ENDPOINTS.accommodations, data, {
+        headers: { 'Content-Type': 'application/json' },
```

Comment:
no need to include this header, it is already in the default config. Following the request of our teacher I also renamed `axiosInstance` to just `api`.